packages/gatsby-theme-aio/gatsby-config.js

```diff
@@ -91,6 +135,26 @@ module.exports = {
         scriptUrl: process.env.ADOBE_LAUNCH_SRC,
         includeInDevelopment: process.env.ADOBE_LAUNCH_SRC_INCLUDE_IN_DEVELOPMENT || false
       }
+    },
+    {
+      resolve: `gatsby-plugin-algolia`,
+      options: {
+        appId: process.env.ALGOLIA_APP_ID,
```

Comment:
what happens if you dont provide any algolia env values ? Will the plugin crash the build ?

Reply:
@naydav, as we discussed for the other PR, make sure you are account for two scenarios: no Algolia .env variables at all and .env variables without values, such as ALGOLIA_APP_ID= Which evaluates to an empty string for the variable. For both cases, as long as you are not indexing by default, these variables should not come into play because the plug-in only uses them when indexing. 

Again, all this is a non-issue if we remove all Algolia indexing variables from the .env variables at each site and control indexing from the theme itself, where we run it for individual sites as needed. This takes the burden off of site administrators, eliminates all the indexing config errors they will inevitably make, and eliminate the security risks of leaking secrets. 

ACTION ITEM:
So as a part of this and the other PR (DEVSITE-47), we should add this centralization of variables/secrets to the theme and remove the dependencies on .env variables from individual sites.

Reply:
I created separate PR for this https://github.com/adobe/gatsby-theme-aio/pull/400 (that PR include code of this one)